From 936b734d5f14cc0a0bcab9c1bf9b436842ca637e Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Thu, 8 Jan 2026 22:00:15 +0100
Subject: [PATCH] fix: Add fmt 11+ compatibility to PropertyFmt.h

Disable fmt's automatic range formatting for Gaudi::Property to prevent
ambiguous formatter resolution with fmt 11+, which auto-detects ranges.
---
 GaudiKernel/include/Gaudi/PropertyFmt.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/GaudiKernel/include/Gaudi/PropertyFmt.h b/GaudiKernel/include/Gaudi/PropertyFmt.h
index 4022de3d9f..7c296ce3c1 100644
--- a/GaudiKernel/include/Gaudi/PropertyFmt.h
+++ b/GaudiKernel/include/Gaudi/PropertyFmt.h
@@ -1,5 +1,5 @@
 /***********************************************************************************\
-* (c) Copyright 2023-2025 CERN for the benefit of the LHCb and ATLAS collaborations *
+* (c) Copyright 2023-2026 CERN for the benefit of the LHCb and ATLAS collaborations *
 *                                                                                   *
 * This software is distributed under the terms of the Apache version 2 licence,     *
 * copied verbatim in the file "LICENSE".                                            *
@@ -53,10 +53,16 @@
 
 #include <Gaudi/Property.h>
 #include <fmt/format.h>
+#include <fmt/ranges.h>
 #include <format>
 #include <iomanip>
 #include <sstream>
 
+// Disable fmt's automatic range formatting for Gaudi::Property
+// This prevents ambiguity with fmt 11+ which auto-detects ranges
+template <typename T, typename V, typename H, typename Char>
+struct fmt::is_range<Gaudi::Property<T, V, H>, Char> : std::false_type {};
+
 #define GAUDI_PROPERTY_FORMATTING_IMPL( ns, assert_for_ranges )                                                        \
   template <typename T, typename V, typename H>                                                                        \
   struct ns::formatter<Gaudi::Property<T, V, H>> : ns::formatter<T> {                                                  \
-- 
GitLab

